src/plugins/fcrypt/fcrypt.c

```diff
@@ -230,15 +234,29 @@ static int fcryptGpgCallAndCleanup (Key * parentKey, KeySet * pluginConfig, char
 		// gpg call returned success, overwrite the original file with the gpg payload data
 		if (rename (tmpFile, keyString (parentKey)) != 0)
 		{
-			ELEKTRA_SET_RESOURCE_ERRORF (parentKey, "Renaming file %s to %s failed. Reason: %s", tmpFile, keyString (parentKey),
-						     strerror (errno));
-			result = -1;
+			// if rename failed we can still try to copy the file content manually
+			lseek (tmpFileFd, 0, SEEK_SET);
+			lseek (parentKeyFd, 0, SEEK_SET);
+			readCount = read (tmpFileFd, buffer, sizeof (buffer));
+			while (readCount > 0)
+			{
+				writeCount = write (parentKeyFd, buffer, readCount);
+				if (writeCount != readCount)
+				{
+					ELEKTRA_SET_RESOURCE_ERRORF (parentKey, "Data transfer from file %s to %s failed. Reason: %s",
+								     tmpFile, keyString (parentKey), strerror (errno));
+					result = -1;
+					break;
+				}
+				readCount = read (tmpFileFd, buffer, sizeof (buffer));
+			}
+			manualCopy = 1;
 		}
 	}
 
 	if (result == 1)
 	{
-		if (parentKeyFd >= 0)
+		if (parentKeyFd >= 0 && !manualCopy)
```

Comment:
Is there harm in shredding anyway? The fact that rename failed but the content could be copied does not guarantee that it was in ramdisk. I don't know how ramdisk works but it is not a bad idea to also overwrite sensitive data in ram.

Reply:
`parentKeyFd` holds the "target" data, so we would shred the content that we manually copied before. I do not remember why I introduced this shred in the first place. But I think in case of `manualCopy` the `tmpFile` should be unlinked. I will check this tomorrow.